audio/src/fetch.rs

```diff
@@ -334,36 +338,34 @@ impl AudioFileOpenStreaming {
 }
 
 impl Future for AudioFileOpen {
-    type Item = AudioFile;
-    type Error = ChannelError;
+    type Output = Result<AudioFile, ChannelError>;
 
-    fn poll(&mut self) -> Poll<AudioFile, ChannelError> {
+    fn poll(self: Pin<&mut Self>, cx: &mut Context) -> Poll<Result<AudioFile, ChannelError>> {
         match *self {
             AudioFileOpen::Streaming(ref mut open) => {
-                let file = try_ready!(open.poll());
-                Ok(Async::Ready(AudioFile::Streaming(file)))
+                let file = ready!(open.compat().poll());
+                Poll::Ready(Ok(AudioFile::Streaming(file)))
             }
             AudioFileOpen::Cached(ref mut file) => {
                 let file = file.take().unwrap();
-                Ok(Async::Ready(AudioFile::Cached(file)))
+                Poll::Ready(Ok(AudioFile::Cached(file)))
             }
         }
     }
 }
 
 impl Future for AudioFileOpenStreaming {
```

Comment:
If you wanted you could probably get rid of a good deal of these manually impl'd futures.

```
async fn open(&mut self) -> Result<Self, ChannelError> {
   let (id, data) = self.headers.await?;
  ...
}
```